src/metrics/metrics.jl

```diff
@@ -88,123 +94,133 @@ total_absolute_cover = Metric(
-    end
-    n_sc::Int64 = Int64(n_group_and_size / n_groups)
+)::AbstractArray{<:Real,2}
+    n_timesteps, n_group_sizes, n_locs = size(X)
+    _n_sizes::Int64 = n_sizes(n_groups, n_group_sizes)
 
-    taxa_cover::YAXArray = ZeroDataCube((:timesteps, :taxa), (n_steps, n_groups))
-    k_cover = zeros(n_steps, n_sc, n_locs)
-    for (taxa_id, grp) in enumerate([i:(i + (n_sc - 1)) for i in 1:n_sc:n_group_and_size])
-        for (loc, a) in enumerate(k_area)
-            k_cover[:, :, loc] .= X[:, grp, loc] .* a
+    taxa_cover::YAXArray = ZeroDataCube(
+        (:timesteps, :species), (n_timesteps, n_groups), X.properties
+    )
+    k_cover = zeros(n_timesteps, _n_sizes, n_locs)
+    _group_indices::Vector{UnitRange{Int64}} = group_indices(_n_sizes, n_group_sizes)
+    for (idx_group, group) in enumerate(_group_indices)
+        # Fill k_cover with absolute cover for each timestep and location and fixed group
+        for (idx_loc, loc_k_area) in enumerate(k_area)
+            k_cover[:, :, idx_loc] .= X[:, group, idx_loc] .* loc_k_area
         end
 
-        # Sum over size class groups
-        taxa_cover[:, taxa_id] = vec(sum(k_cover; dims=(2, 3))) ./ sum(k_area)
+        # Sum over size classes and locations  and divide by total k area
+        taxa_cover[:, idx_group] = vec(sum(k_cover; dims=(2, 3))) ./ sum(k_area)
     end
 
     return taxa_cover
 end
-function _relative_taxa_cover(rs::ResultSet)::AbstractArray
+function _relative_taxa_cover(rs::ResultSet)::AbstractArray{<:Real,3}
     return rs.outcomes[:relative_taxa_cover]
 end
-relative_taxa_cover = Metric(_relative_taxa_cover, (:timesteps, :taxa, :scenarios))
+relative_taxa_cover = Metric(_relative_taxa_cover, (:timesteps, :species, :scenarios))
 
 """
-    relative_loc_taxa_cover(X::AbstractArray{T}, k_area::Vector{T}, n_groups::Int64)::AbstractArray where {T<:Real}
+    relative_loc_taxa_cover(X::AbstractArray{T}, k_area::Vector{T}, n_groups::Int64)::AbstractArray{T,3} where {T<:Real}
+
+# Arguments
+- `X` : Raw model results for a single scenario. Dimensions (n_timesteps, n_group_sizes,
+n_locations)
+- `k_area` : The coral habitable area.
+- `n_groups` : Number of function coral groups.
+
+# Returns
+Coral cover, grouped by taxa for the given scenario, for each timestep and location,
+relative to location k area.
 """
 function _relative_loc_taxa_cover(
     X::AbstractArray{T}, k_area::Vector{T}, n_groups::Int64
-)::AbstractArray where {T<:Real}
-    n_steps, n_group_and_size, n_locs = size(X)
-    if n_group_and_size % n_groups != 0
-        throw(
-            ArgumentError(
-                "Number of groups must divide n_group_and_size. n_group_and_size $(n_group_and_size), n_groups: $(n_groups)"
-            )
-        )
-    end
-    n_sc::Int64 = Int64(n_group_and_size / n_groups)
+)::AbstractArray{T,3} where {T<:Real}
+    n_timesteps, n_group_sizes, n_locs = size(X)
+    _n_sizes::Int64 = n_sizes(n_groups, n_group_sizes)
 
     taxa_cover::YAXArray = ZeroDataCube(
```

Comment:
Maybe in the future, but since the metric is currently called `relative_loc_taxa_cover`, we would need to change the metric name as well - which would be good, but maybe not in this PR